app/models/discussion_lock.rb

```diff
@@ -0,0 +1,17 @@
+class DiscussionLock < ApplicationRecord
+  belongs_to :article
+  belongs_to :locking_user, class_name: "User"
+
+  before_validation :nullify_blank_notes_and_reason
+
+  validates :article_id, presence: true, uniqueness: true
+  validates :locking_user_id, presence: true
+
+  private
+
+  def nullify_blank_notes_and_reason
+    # Prevent blank strings from beings saved to the DB
+    self.notes = nil if notes.blank?
+    self.reason = nil if reason.blank?
```

Comment:
What's the reason behind this choice?

To be clear, I agree very much in principle with the idea of not having both `NULL` and empty strings. I just haven't noticed us using this particular idea elsewhere.

Reply:
It's because the `notes` and `reason` fields are optional and if a user doesn't type anything (we expect this to be the most common case) then empty strings (`""`) are passed through and saved.

I don't feel strongly about this - I'm happy to just remove it if folks feel some type of way about this.

Reply:
I know that you're not the biggest fan of metaprogramming @jgaskins but we could really easily share this behavior via a module builder. Here's the quick and dirty 5-minute version:

```ruby
module AttributeCleaner
  def self.for(*attrs)
    Module.new do
      def self.included(klass)
        klass.before_validation :nullify_blank_attributes
      end

      define_method :nullify_blank_attributes do
        attrs.each { |attr| public_send("#{attr}=", nil) if public_send(attr).blank? }
      end
    end
  end
end
```

You then include it like `include AttributeCleaner.for(:attr1, :attr2).

Here it is in action (in a Rails console where I loaded the code above):

```ruby
Profile.include(AttributeCleaner.for(:location))
p = Profile.new(location: "    ")
p.valid? 
#=> false
p.location
#=> nil
```